src/App.tsx

```diff
@@ -116,75 +126,97 @@ const hexRegex = /^#[a-f0-9]{6}$/gi;
 
 // TODO: add an error boundary for parse fails
 const Editor: Component<{ currentFile: FileSystemFileHandle }> = (props) => {
-  const [contents, { mutate }] = createResource(async () => {
+  const [obj, setObject] = createStore<Record<string, unknown>>({});
+  let lastUpdate = Date.now();
+  createEffect(async () => {
```

Comment:
I think we can get away with not using an effect here? see https://www.solidjs.com/tutorial/lifecycles_oncleanup?solved

I might be wrong, though. this might be necessary to detect changes in the props?